packages/app/src/cli/services/dev/extension/server/utilities.ts

```diff
@@ -67,6 +67,7 @@ function getCustomerAccountsRedirectUrl(
   rawUrl.searchParams.append('origin', origin)
   rawUrl.searchParams.append('extensionId', extension.devUUID)
   rawUrl.searchParams.append('source', 'CUSTOMER_ACCOUNT_EXTENSION')
+  rawUrl.searchParams.append('appId', options.id ?? '')
```

Comment:
Do you know when `options.id` might not exist?

Reply:
To be honest, I don't know why that would be the case, but it's [a nilable type](https://github.com/Shopify/cli/blob/dda4d433c25e1c9bb6b8426280707e3c699be841/packages/app/src/cli/services/dev/extension.ts#L36-L39). Maybe that was done to support a specific use case, but for us, I'm not sure how we would have an extension without a corresponding app.

Reply:
It was introduced [here](https://github.com/Shopify/cli/pull/958) so maybe it was just made nilable since it populates a new property added to an existing model and they didn't want to make it required?